charts/konk/scripts/provision.sh

```diff
@@ -0,0 +1,17 @@
+kubeadm init phase certs all
+kubeadm init phase kubeconfig admin
+VAL=$(kubectl -n {{ .Release.Namespace }} get secret {{ include "konk.fullname" . }}-etcd-cert --ignore-not-found)
+if [[ $VAL == "" ]]
```

Comment:
Instead of string matching, you can just use the exit code.
```suggestion
if ! kubectl -n {{ .Release.Namespace }} get secret {{ include "konk.fullname" . }}-etcd-cert
```

Reply:
Speaking of which, we should probably improve the script to error out when `kubectl` requests fail.

Reply:
If you take off `--ignore-not-found` exclude `-e` from the setopts above

Reply:
No, you still want the `-e`. Since it's in an `if`, it won't cause the whole script to fail. Demo:
```
thayward@tac-ml-thayward ~ % cat test.sh                                                                                                                                       env-4
set -xe
if ! kubectl get ns foobar
then
	echo true
fi
thayward@tac-ml-thayward ~ % ./test.sh                                                                                                                                         env-4
+ kubectl get ns foobar
Error from server (NotFound): namespaces "foobar" not found
+ echo true
true
```

Reply:
Updated to use exit code instead of string matching.